File: app/src/main/java/com/example/belajar_retrofit/datamodels/ListLogbookResponse.java
package com.example.belajar_retrofit.datamodels;

import java.util.List;

public class ListLogbookResponse{
	private String reportTitle;
	private String endAt;
	private Object linkSeminar;
	private int proposalId;
	private Object attendeesList;
	private int supervisorId;
	private Object newsEvent;
	private int seminarRoomId;
	private Object certificate;
	private List<LogbooksItem> logbooks;
	private int studentId;
	private Object createdAt;
	private Object startAt;
	private Object cancellationReason;
	private Object internshipScore;
	private String seminarDate;
	private Object reportReceipt;
	private String updatedAt;
	private String grade;
	private Object workReport;
	private Object seminarDeadline;
	private int id;
	private Object activityReport;
	private int status;

	public String getReportTitle(){
		return reportTitle;
	}

	public String getEndAt(){
		return endAt;
	}

	public Object getLinkSeminar(){
		return linkSeminar;
	}

	public int getProposalId(){
		return proposalId;
	}

	public Object getAttendeesList(){
		return attendeesList;
	}

	public int getSupervisorId(){
		return supervisorId;
	}

	public Object getNewsEvent(){
		return newsEvent;
	}

	public int getSeminarRoomId(){
		return seminarRoomId;
	}

	public Object getCertificate(){
		return certificate;
	}

	public List<LogbooksItem> getLogbooks(){
		return logbooks;
	}

	public int getStudentId(){
		return studentId;
	}

	public Object getCreatedAt(){
		return createdAt;
	}

	public Object getStartAt(){
		return startAt;
	}

	public Object getCancellationReason(){
		return cancellationReason;
	}

	public Object getInternshipScore(){
		return internshipScore;
	}

	public String getSeminarDate(){
		return seminarDate;
	}

	public Object getReportReceipt(){
		return reportReceipt;
	}

	public String getUpdatedAt(){
		return updatedAt;
	}

	public String getGrade(){
		return grade;
	}

	public Object getWorkReport(){
		return workReport;
	}

	public Object getSeminarDeadline(){
		return seminarDeadline;
	}

	public int getId(){
		return id;
	}

	public Object getActivityReport(){
		return activityReport;
	}

	public int getStatus(){
		return status;
	}
}

File: app/src/main/java/com/example/belajar_retrofit/datamodels/ListLogbookResponse.kt
package com.example.belajar_retrofit.datamodels

import com.google.gson.annotations.SerializedName

data class ListLogbookResponse(

	@field:SerializedName("count")
	val count: Int? = null,

	@field:SerializedName("logbooks")
	val logbooks: List<LogbooksItem?>? = null,

	@field:SerializedName("status")
	val status: String? = null
)

data class LogbooksItem(

	@field:SerializedName("date")
	val date: String? = null,

	@field:SerializedName("thesis_id")
	val thesisId: Int? = null,

	@field:SerializedName("notes")
	val notes: Any? = null,

	@field:SerializedName("supervisor_id")
	val supervisorId: Int? = null,

	@field:SerializedName("created_at")
	val createdAt: String? = null,

	@field:SerializedName("supervised_at")
	val supervisedAt: Any? = null,

	@field:SerializedName("file_progress")
	val fileProgress: String? = null,

	@field:SerializedName("supervised_by")
	val supervisedBy: Any? = null,

	@field:SerializedName("problem")
	val problem: String? = null,

	@field:SerializedName("updated_at")
	val updatedAt: String? = null,

	@field:SerializedName("progress")
	val progress: String? = null,

	@field:SerializedName("file_notes")
	val fileNotes: Any? = null,

	@field:SerializedName("id")
	val id: Int? = null,

	@field:SerializedName("status")
	val status: Int? = null
)
